src/main/scala/vpio/IO.scala

```diff
@@ -60,25 +128,51 @@ object IO extends CommandApp(
       val layout: LayoutDefinition =
         ZoomedLayoutScheme.layoutForZoom(14, WebMercator.worldExtent, 512)
 
-      osm.fromORC(orc) match {
+      val df = ss.read.orc(orc)
+      //      val df = ss.read.orc("s3://osm-pds/planet/planet-latest.orc")
+
+      // val southAmericaBeforeJune =
+      //   df.where("timestamp <= '2017-06-01 00:00:00.0' AND (type != 'node' OR (lon <= -34.767608642578125 AND lon >= -81.33865356445312 AND lat <= 12.623252653219012 AND lat >= -55.98609153380838))")
+
+      // val southAmerica =
+      //   df.where("type != 'node' OR (lon <= -34.767608642578125 AND lon >= -81.33865356445312 AND lat <= 12.623252653219012 AND lat >= -55.98609153380838)")
+
+      val targetDf = df//.repartition(1000)
+
+      // Test log clip fail
+      val ff = Feature(Point(1,1), vectorpipe.util.Tree(osm.ElementData(osm.ElementMeta(1L, "Asdf", "asdf", 2L, 3L, "asdf", true), Map(), None), Seq(vectorpipe.util.Tree(osm.ElementData(osm.ElementMeta(2L, "Asdf", "asdf", 2L, 3L, "asdf", true), Map(), None), Seq()))))
+
+      Util.logClipFail(Extent(0, 0, 1, 1), ff)
+
+
+      osm.fromDataFrame(targetDf) match {
```

Comment:
Generating from a dataframe allows us to do some spark sql up top.

I want to be able to do a dataframe per type, node way and relation.